src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1025,12 +1025,15 @@ public static bool IsValidParams(Symbol member)
                 return false;
             }
 
-            // Note: we need to confirm the "arrayness" on the original definition because
-            // it's possible that the type becomes an array as a result of substitution.
             ParameterSymbol final = member.GetParameters().Last();
```

Comment:
>member.GetParameters().Last() [](http://example.com/codeflow?start=36&length=29)

I think this important part should be part of ```IsValidParamsParameter```, i.e. only the last parameter can be a valid params parameter.